src/data_conn.c

```diff
@@ -335,6 +370,7 @@ handle_data_conn_error(replica_t *r)
 		pthread_cond_signal(&r->r_cond);
 	r->mgmt_eventfd2 = -1;
 	MTX_UNLOCK(&r->r_mtx);
+	destroy_mempool(&r->cmdq);
```

Comment:
lets move this to free_replica and also check for any other left out allocations..